components/script/dom/html/htmlinputelement.rs

```diff
@@ -345,23 +345,27 @@ impl InputType {
             InputType::Week => "week",
         }
     }
+}
 
-    pub(crate) fn as_ime_type(&self) -> Option<InputMethodType> {
-        match *self {
-            InputType::Color => Some(InputMethodType::Color),
-            InputType::Date => Some(InputMethodType::Date),
-            InputType::DatetimeLocal => Some(InputMethodType::DatetimeLocal),
-            InputType::Email => Some(InputMethodType::Email),
-            InputType::Month => Some(InputMethodType::Month),
-            InputType::Number => Some(InputMethodType::Number),
-            InputType::Password => Some(InputMethodType::Password),
-            InputType::Search => Some(InputMethodType::Search),
-            InputType::Tel => Some(InputMethodType::Tel),
-            InputType::Text => Some(InputMethodType::Text),
-            InputType::Time => Some(InputMethodType::Time),
-            InputType::Url => Some(InputMethodType::Url),
-            InputType::Week => Some(InputMethodType::Week),
-            _ => None,
+impl TryInto<InputMethodType> for InputType {
```

Comment:
The std library recommends we implement `TryFrom<InputType> for InputMethodType` rather than `TryInto`.

https://doc.rust-lang.org/std/convert/trait.TryInto.html